cmake/FindGMP.cmake

```diff
@@ -9,17 +9,80 @@
 ## directory for licensing information.
 ##
 # Find GMP
-# GMP_FOUND - system has GMP lib
-# GMP_INCLUDE_DIR - the GMP include directory
-# GMP_LIBRARIES - Libraries needed to use GMP
+# GMP_FOUND - should always be true
+# GMP - target for the GMP library
+
+include(deps-helper)
 
 find_path(GMP_INCLUDE_DIR NAMES gmp.h gmpxx.h)
 find_library(GMP_LIBRARIES NAMES gmp)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(GMP DEFAULT_MSG GMP_INCLUDE_DIR GMP_LIBRARIES)
+set(GMP_FOUND_SYSTEM FALSE)
+if(GMP_INCLUDE_DIR AND GMP_LIBRARIES)
+  set(GMP_FOUND_SYSTEM TRUE)
+
+  function(GetVersionPart OUTPUT FILENAME DESC)
+    file(STRINGS ${FILENAME} RES REGEX "^#define __GNU_MP_${DESC}[ \\t]+.*")
+    string(REGEX MATCH "[0-9]+" RES "${RES}")
+    set(${OUTPUT}
+        "${RES}"
+        PARENT_SCOPE
+    )
+  endfunction()
+  function(GetVersion OUTPUT FILENAME)
+    getversionpart(MAJOR "${FILENAME}" "VERSION")
```

Comment:
Either use `GetVersionPart` here or rename the function to `getversionpart` (just for consistency).